src/components/application_manager/include/application_manager/application_manager_impl.h

```diff
@@ -1443,7 +1446,7 @@ class ApplicationManagerImpl
    * will send TTS global properties to HMI after timeout
    */
   std::map<uint32_t, TimevalStruct> tts_global_properties_app_list_;
-
+  std::set<connection_handler::DeviceHandle> query_apps_devices_;
```

Comment:
@Ypostolov Can occurred data race with the filed 'query_apps_devices_'?

Reply:
@Ypostolov my opinion it should be protected.

Reply:
@EKuliiev @AByzhynar 
Fixed in e691feb95fa38c5a9082ca7e2fb1d1e2add1eccb